3.16.0 build fails with -DWITH_BINDINGS=no && WITH_QGIS_PROCESS=yes in target `qgis_process`

Even though there is a conditional to be found inside `src/process/CMakeLists.txt`...
```
IF (WITH_BINDINGS)
  TARGET_LINK_LIBRARIES(qgis_process qgispython)
ENDIF(WITH_BINDINGS)
```

...this doesn't build as `main.cpp` and `qgsprocess.cpp` unconditionally include `qgsprocess.h` which then unconditionally includes `qgspythonrunner.h` and `qgspythonutils.h`.


Used build options:
```
cmake -G Ninja -DCMAKE_INSTALL_PREFIX=/usr -DQGIS_MANUAL_SUBDIR=share/man/ -DQGIS_LIB_SUBDIR=lib64 \
-DQGIS_PLUGIN_SUBDIR=lib64/qgis -DQWT_INCLUDE_DIR=/usr/include/qwt6 -DQWT_LIBRARY=/usr/lib64/libqwt6-qt5.so \
-DPEDANTIC=OFF -DUSE_CCACHE=OFF -DWITH_ANALYSIS=ON -DWITH_APIDOC=OFF -DWITH_GUI=ON -DWITH_INTERNAL_MDAL=ON \
-DWITH_QSPATIALITE=ON -DENABLE_TESTS=OFF -DWITH_3D=no -DWITH_GEOREFERENCER=no -DWITH_GRASS7=no \
-DCMAKE_DISABLE_FIND_PACKAGE_HDF5=ON -DWITH_SERVER=no -DCMAKE_DISABLE_FIND_PACKAGE_NetCDF=ON -DUSE_OPENCL=no \
-DWITH_ORACLE=no -DWITH_QWTPOLAR=no -DWITH_POSTGRESQL=yes -DWITH_BINDINGS=no -DWITH_CUSTOM_WIDGETS=no \
-DWITH_QUICK=yes -DWITH_QTWEBKIT=OFF
```
Error snippet:
```
In file included from src/process/qgis_process_autogen/EWIEGA46WW/../../../../../qgis-3.16.0/src/process/qgsprocess.h:22,
                 from src/process/qgis_process_autogen/EWIEGA46WW/moc_qgsprocess.cpp:10,
                 from src/process/qgis_process_autogen/mocs_compilation.cpp:2:
qgis-3.16.0/src/python/qgspythonutils.h:22:10: fatal error: qgis_python.h: No such file or directory
   22 | #include "qgis_python.h"
      |          ^~~~~~~~~~~~~~~
compilation terminated.
```

--- a/src/process/CMakeLists.txt
+++ b/src/process/CMakeLists.txt
@@ -21,16 +21,10 @@ add_executable (qgis_process MACOSX_BUNDLE ${QGIS_PROCESS_SRCS})
 # require c++17
 target_compile_features(qgis_process PRIVATE cxx_std_17)
 
-include_directories(
+target_include_directories(qgis_process PUBLIC
   ${CMAKE_CURRENT_BINARY_DIR}
 )
-include_directories(SYSTEM
-  ${QT_INCLUDE_DIR}
-  ${QGIS_INCLUDE_DIR}
-  ${PROJ_INCLUDE_DIR}
-  ${GDAL_INCLUDE_DIR}
-  ${GEOS_INCLUDE_DIR}
-)
+
 
 if (UNIX AND NOT ANDROID)
   include_directories(SYSTEM
@@ -38,23 +32,10 @@ if (UNIX AND NOT ANDROID)
   )
 endif()
 
-include_directories(
-  ${CMAKE_SOURCE_DIR}/src/python
-
-  ${CMAKE_BINARY_DIR}/src/python
-  ${CMAKE_BINARY_DIR}/src/analysis
-)
-
 if (WITH_3D)
-  include_directories(
-  )
   include_directories(SYSTEM
     ${QT5_3DEXTRA_INCLUDE_DIR}
   )
-endif()
-
-if (WITH_BINDINGS)
-  target_link_libraries(qgis_process qgispython)
 endif()
 
 target_link_libraries(qgis_process
@@ -70,6 +51,11 @@ if (WITH_3D)
   target_link_libraries(qgis_process
     qgis_3d
   )
+endif()
+
+if (WITH_BINDINGS)
+  target_link_libraries(qgis_process qgispython)
+  add_definitions(-DWITH_BINDINGS)
 endif()
 
 # install

--- a/src/process/qgsprocess.cpp
+++ b/src/process/qgsprocess.cpp
@@ -174,7 +174,7 @@ void ConsoleFeedback::showTerminalProgress( double progress )
     fflush( stdout );
 }
 
-
+#ifdef WITH_BINDINGS
 //! load Python support if possible
 std::unique_ptr< QgsPythonUtils > QgsProcessingExec::loadPythonSupport()
 {
@@ -218,6 +218,7 @@ std::unique_ptr< QgsPythonUtils > QgsProcessingExec::loadPythonSupport()
 
   return pythonUtils;
 }
+#endif
 
 QgsProcessingExec::QgsProcessingExec()
 {
@@ -264,6 +265,8 @@ int QgsProcessingExec::run( const QStringList &constArgs )
 #ifdef HAVE_3D
   QgsApplication::processingRegistry()->addProvider( new Qgs3DAlgorithms( QgsApplication::processingRegistry() ) );
 #endif
+
+#ifdef WITH_BINDINGS
 
   // give Python plugins a chance to load providers
   mPythonUtils = loadPythonSupport();
@@ -447,6 +450,7 @@ int QgsProcessingExec::run( const QStringList &constArgs )
   {
     std::cerr << QStringLiteral( "Command %1 not known!\n" ).arg( command ).toLocal8Bit().constData();
   }
+#endif
   return 1;
 }
 
@@ -475,6 +479,7 @@ void QgsProcessingExec::showUsage( const QString &appName )
   std::cout << msg.join( QString() ).toLocal8Bit().constData();
 }
 
+#ifdef WITH_BINDINGS
 void QgsProcessingExec::loadPlugins()
 {
   QgsSettings settings;
@@ -495,6 +500,7 @@ void QgsProcessingExec::loadPlugins()
     }
   }
 }
+#endif
 
 void QgsProcessingExec::listAlgorithms( bool useJson )
 {
@@ -578,6 +584,7 @@ void QgsProcessingExec::listPlugins( bool useJson, bool showLoaded )
     addVersionInformation( json );
   }
 
+#ifdef WITH_BINDINGS
   QVariantMap jsonPlugins;
   const QStringList plugins = mPythonUtils->pluginList();
   for ( const QString &plugin : plugins )
@@ -606,6 +613,7 @@ void QgsProcessingExec::listPlugins( bool useJson, bool showLoaded )
     json.insert( QStringLiteral( "plugins" ), jsonPlugins );
     std::cout << QgsJsonUtils::jsonFromVariant( json ).dump( 2 );
   }
+#endif
 }
 
 int QgsProcessingExec::enablePlugin( const QString &name, bool enabled )
@@ -615,6 +623,7 @@ int QgsProcessingExec::enablePlugin( const QString &name, bool enabled )
   else
     std::cout << QStringLiteral( "Disabling plugin: \"%1\"\n" ).arg( name ).toLocal8Bit().constData();
 
+#ifdef WITH_BINDINGS
   const QStringList plugins = mPythonUtils->pluginList();
   if ( !plugins.contains( name ) )
   {
@@ -664,6 +673,10 @@ int QgsProcessingExec::enablePlugin( const QString &name, bool enabled )
   listPlugins( false, false );
 
   return 0;
+#else
+  std::cerr << "No Python support\n";
+  return 1;
+#endif
 }
 
 int QgsProcessingExec::showAlgorithmHelp( const QString &id, bool useJson )

--- a/src/process/qgsprocess.h
+++ b/src/process/qgsprocess.h
@@ -17,9 +17,12 @@
 #ifndef QGSPROCESS_H
 #define QGSPROCESS_H
 
-#include "qgsprocessingfeedback.h"
+#ifdef WITH_BINDINGS
 #include "qgspythonrunner.h"
 #include "qgspythonutils.h"
+#endif
+
+#include "qgsprocessingfeedback.h"
 #include "qgsunittypes.h"
 #include "qgsprocessingcontext.h"
 #include <QElapsedTimer>
@@ -90,8 +93,11 @@ class QgsProcessingExec
     void addAlgorithmInformation( QVariantMap &json, const QgsProcessingAlgorithm *algorithm );
     void addProviderInformation( QVariantMap &json, QgsProcessingProvider *provider );
 
+
+#ifdef WITH_BINDINGS
     std::unique_ptr< QgsPythonUtils > mPythonUtils;
     std::unique_ptr<QgsPythonUtils> loadPythonSupport();
+#endif
 };
 
 #endif // QGSPROCESS_H

